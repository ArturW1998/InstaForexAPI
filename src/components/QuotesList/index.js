import React from 'react';
import { Text, FlatList, View, ActivityIndicator } from 'react-native';
import useDataApi from '../../hooks/useDataApi';
import Quote from '../Quote';

const QuotesList = () => {
    const [{ data, isLoading, isError }] = useDataApi();

    const renderItem = ({ item }) => <Quote symbol={item.symbol} />;
    const keyExtractor = item => item.symbol;

    return (
        <View>
            {isError && <Text>Something went wrong...</Text>}

            {isLoading ? (
                <ActivityIndicator size="large" color="#0000ff" />
            ) : (
                <FlatList
                    data={data}
                    renderItem={renderItem}
                    keyExtractor={keyExtractor}
                />
            )}
        </View>
    );
};

export default QuotesList;
