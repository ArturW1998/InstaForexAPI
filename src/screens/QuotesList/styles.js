import { StyleSheet } from "react-native";
import { deviceHeight, deviceWidth } from "../../services/getDeviceSize";

const styles = StyleSheet.create({
    container: {
        marginTop: 10,
        flex: 1,
        alignItems: 'center'
    },
    quotes: {
        borderColor: 'gray',
        borderWidth: 1,
        backgroundColor: '#fff',
        borderRadius: 5,
        padding: 10,
        width: deviceWidth,
        height: deviceHeight - 270
    },
});

export default styles;
