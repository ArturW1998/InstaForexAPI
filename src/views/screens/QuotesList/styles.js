import { StyleSheet } from 'react-native';
import { LIGHT_COLOR, MUTED_COLOR } from '../../../constants/themes';
import { deviceHeight, deviceWidth } from '../../../services/getDeviceSize';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    marginTop: 10,
  },
  quotes: {
    backgroundColor: LIGHT_COLOR,
    borderColor: MUTED_COLOR,
    borderRadius: 5,
    borderWidth: 1,
    height: deviceHeight - 270,
    padding: 10,
    width: deviceWidth,
  },
});

export default styles;
