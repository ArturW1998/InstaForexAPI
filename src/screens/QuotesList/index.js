import React, { useContext } from 'react';
import { FlatList, View } from 'react-native';
import PropTypes from 'prop-types';
import { DETAILS_SCREEN } from '../../constants/routes'
import { DANGER_COLOR } from "../../constants/themes";
import { QuotesListContext } from '../../contextes/quotesListContext';
import ErrorIndicator from "../../components/ErrorIndicator";
import AppButton from "../../components/AppButton";
import AppLoader from "../../components/AppLoader";
import EmptyPage from "../../components/EmptyPage";
import Search from "../../components/Search";
import Pagination from "../../components/Pagination";
import Quote from '../../components/Quote';
import styles from './styles';

const QuotesList = ({ navigation }) => {
    const { data, isError, isLoading, currentPage, totalPages, search, updateSearch, fetchData, paginate } = useContext(QuotesListContext);

    const openDetails = (symbol, description, digits) => navigation.navigate(DETAILS_SCREEN, { symbol, description, digits });

    const renderItem = ({ item: { symbol, description, digits } }) => (
        <Quote symbol={symbol} description={description} digits={digits} onPress={openDetails} />
    );
    const keyExtractor = ({ symbol }) => symbol;
    const renderEmpty = () => <EmptyPage />;

    if (isError) {
        return (
            <ErrorIndicator>
                <AppButton backgroundColor={DANGER_COLOR} onPress={fetchData}>Try again</AppButton>
            </ErrorIndicator>
        )
    }

    if (isLoading) return <AppLoader />;

    return (
        <View style={styles.container}>
            <Search onChangeText={updateSearch} value={search} />

            <View style={styles.quotes}>
                <FlatList
                    data={data}
                    onRefresh={fetchData}
                    refreshing={isLoading}
                    renderItem={renderItem}
                    keyExtractor={keyExtractor}
                    ListEmptyComponent={renderEmpty}
                />
            </View>

            <Pagination
                totalPages={totalPages}
                currentPage={currentPage}
                onPress={paginate}
            />
        </View>
    );
};

QuotesList.propTypes = {
    navigation: PropTypes.shape({
        navigate: PropTypes.func,
    }),
};

QuotesList.defaultProps = {
    navigation: {
        navigate: () => {},
    },
};

export default QuotesList;
