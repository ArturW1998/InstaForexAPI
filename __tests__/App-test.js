import React from 'react';
import renderer from 'react-test-renderer';
import QuotesList from '../src/screens/QuotesList';

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<QuotesList />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
