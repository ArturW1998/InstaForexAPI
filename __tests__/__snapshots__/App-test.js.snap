// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginTop": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search..."
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#808080",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "fontSize": 26,
        "justifyContent": "space-between",
        "marginBottom": 10,
        "padding": 10,
        "width": 730,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "#808080",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 1064,
        "padding": 10,
        "width": 730,
      }
    }
  >
    <RCTScrollView
      ListEmptyComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 300,
              "justifyContent": "center",
              "padding": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "resizeMode": "contain",
                "width": "100%",
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
